src/logger.ts

```diff
@@ -60,3 +62,79 @@ export function sendCrashResponse({
     callback();
   }
 }
+
+export function loggingHandlerAddExecutionContext() {
+  const originalStdoutWrite = process.stdout.write;
+  process.stdout.write = (data, ...args) => {
+    const {encoding, cb} = splitArgs(args);
+    const modifiedData = getModifiedData(data, encoding);
+    return originalStdoutWrite.apply(process.stdout, [modifiedData, cb]);
+  };
+
+  const originalStderrWrite = process.stderr.write;
+  process.stderr.write = (data, ...args) => {
+    const {encoding, cb} = splitArgs(args);
+    const modifiedData = getModifiedData(data, encoding, true);
+    return originalStderrWrite.apply(process.stderr, [modifiedData, cb]);
+  };
+}
+
+export function splitArgs(args: any) {
+  let encoding, cb;
+  if (
+    args.length > 0 &&
+    (Buffer.isEncoding(args[0]) || typeof args[0] === 'undefined')
+  ) {
+    encoding = args[0];
+    args.shift();
+  }
+  if (args.length > 0 && typeof args[0] === 'function') {
+    cb = args[0];
+  }
+  return {encoding: encoding, cb: cb};
+}
+
+export function getModifiedData(
+  data: Uint8Array | string,
+  encoding?: BufferEncoding,
+  stderr = false
+) {
+  let modifiedData = data;
+  if (getCurrentContext()) {
+    const exeuctionContext = getCurrentContext() as ExeuctionContext;
+    const {isJSON, processdData} = processData(data, encoding);
+    let dataWithContext;
+    if (isJSON) {
+      dataWithContext = {...exeuctionContext, ...processdData};
+    } else {
+      dataWithContext = {...exeuctionContext, message: processdData};
+    }
+    if (stderr) {
+      dataWithContext['severity'] = 'ERROR';
+    }
+
+    modifiedData = JSON.stringify(dataWithContext) + '\n';
+  }
+  return modifiedData;
+}
+
+function processData(data: Uint8Array | string, encoding?: BufferEncoding) {
+  let decodedData;
+  try {
+    if (data instanceof Uint8Array) {
+      const textDecoder = new TextDecoder('utf-8');
```

Comment:
I wonder if we can avoid the use of TextDecoder all together. Would the answer 4 from top to down with 35 upvotes of https://stackoverflow.com/questions/8936984/uint8array-to-string-in-javascript help?

Reply:
Done.